presto-main/src/test/java/com/facebook/presto/operator/OperatorAssertion.java

```diff
@@ -192,15 +197,14 @@ public static void assertOperatorEqualsIgnoreOrder(
         assertEqualsIgnoreOrder(actual.getMaterializedRows(), expected.getMaterializedRows());
     }
 
-    static <T> List<T> without(List<T> types, List<Integer> channels)
+    static <T> List<T> without(List<T> list, Collection<Integer> indexes)
     {
-        types = new ArrayList<>(types);
-        int removed = 0;
-        for (int hashChannel : channels) {
-            types.remove(hashChannel - removed);
-            removed++;
-        }
-        return ImmutableList.copyOf(types);
+        Set<Integer> indexesSet = ImmutableSet.copyOf(indexes);
+
+        return IntStream.range(0, list.size())
+                .filter(index -> !indexesSet.contains(index))
+                .mapToObj(ImmutableList.copyOf(list)::get)
```

Comment:
Drop this `ImmutableList.copyOf` wrapper. It reduces clarity and we do expect random-access lists here anyway.